pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2120,6 +2115,15 @@ public TopicStatsImpl getStats(boolean getPreciseBacklog, boolean subscriptionBa
         return statFuture;
     }
 
+    public Optional<CompactedTopicImpl.CompactedTopicContext> getCompactedTopicContext() {
```

Comment:
exposing something named *Impl is a code smell.

can we create an interface for `CompactedTopicContext` ?

Reply:
Yes.  I have made  CompactedTopicContext as a public class.